benches/README.md

```diff
@@ -0,0 +1,71 @@
+# Miden VM Hash Functions 
+In Miden VM, we make use of different hash functions. Some of these are "traditional" hash functions, like `BLAKE3`, which are optimized for out of STARK performance, while others are algebraic hash functions, like `Rescue Prime`, and are more optimized for a better performance inside the STARK. In what follows, we benchmark several such hash functions and compare against other constructions that are used by other proving systems. More precisely, we benchmark:
+
+* **Rescue Prime:**
```

Comment:
Should we spend two sentences on each function to describe it? Like Bobbin does here https://github.com/novifinancial/winterfell/tree/main/crypto

Reply:
Probably not in this README.md

Reply:
I agree, I don't think we need to describe much more except for maybe Poseidon. For Poseidon we already say that is not using vectorized instructions - but are there any other differences?

Reply:
That's the only difference. We can add that the MDS multiplication routine was substituted with its non-vectorized equivalent but I think that should be understood from what was said before.

Reply:
And the field arithmetic implementation is the same as what we have or different?

Reply:
The field arithmetic is different. We use the implementation that was used before the switch to Montgomery.

Reply:
Ah nice! I think that one was basically the same as the one as Polygon Zero uses.

One thing I wonder if we should make it even more specific and say something like:

> As specified [here](https://eprint.iacr.org/2019/458.pdf) and implemented [here](https://github.com/mir-protocol/plonky2/blob/main/plonky2/src/hash/poseidon_goldilocks.rs) (in pure Rust, without vectorized instructions).

To highlight the fact that this implementation is not using any inline assembly either.

Reply:
Or maybe we should make this clear for all arithmetization-friendly hashes (RP, RPO, Poseidon)?

Reply:
Sounds like a great idea, the only thing is that we will have to remove one `asm!` [here](https://github.com/Dominik1999/winterfell/blob/1b434bfb0ac5bce3d0957509b0b6448e4ff91150/math/src/field/f64/mod.rs#L667) .
This actually doesn't affect performance on my laptop so we can just change it so that the code is pure Rust.

Reply:
> Or maybe we should make this clear for all arithmetization-friendly hashes (RP, RPO, Poseidon)?

Basically, we have the following:

1. RPO uses Montgomery representation.
2. RP and Poseidon uses the canonical representation.

We can make mention this if you think it is a good idea.

Reply:
I think this might be too detailed. I did want the fact that it is pure Rust called out so that it is clear that comparison between them is "fair" - but other than that, I think whether it is Montgomery or canonical representation is an implementation detail.

Reply:
Ok, then I will change the Poseidon implementation to be pure Rust and mention this in the docs. As for the details regarding the field implementation, we can just keep it as it is.